src/engine/world/world.h

```diff
@@ -292,6 +292,15 @@ class World final
 
   void addPickupWidget(Sprite sprite, size_t count)
   {
+    auto isSameItemAndSameAction = [sprite](ui::PickupWidget item) -> bool {
```

Comment:
```suggestion
    auto isSameItemAndSameAction = [&sprite](const ui::PickupWidget& item) {
```
Avoid copies for non-primitive types unless necessary. Also, I think something like `isSameVirginItem` would better express what this predicate is doing.